File: scripts/scrapeEvents.py
import json
import requests
from bs4 import BeautifulSoup
import os

def scrape_events(url):
    # Send a GET request to the URL
    response = requests.get(url)

    # If the GET request is successful, the status code will be 200
    if response.status_code == 200:
        # Get the content of the response
        page_content = response.content

        # Create a BeautifulSoup object and specify the parser
        soup = BeautifulSoup(page_content, 'html.parser')

        # Find all event cards on the page
        eventCards = soup.find_all('div', class_='card')

        # Create an empty list to store the events
        events = []

        # Loop through each event card
        for index, card in enumerate(eventCards):
            # Extract the event name
            eventName = card.find('h2', class_='title').text.strip()

            # Extract the event date
            eventDate = card.find('div', class_='fs-6').text.strip()

            # Extract the event link
            eventLink = card.find('a')['href']

            # Extract the event image
            eventImage = card.find('img', class_='card-img-top')['src']

            # Add the event to the list
            events.append({
                'eventName': eventName,
                'eventDate': eventDate,
                'eventLink': eventLink,
                'eventImage': eventImage
            })

        # Return the events
        return events

    else:
        print(f"Failed to retrieve page. Status code: {response.status_code}")
        return None

# Usage
url = "https://www.ministryoftesting.com/events/list"
events = scrape_events(url)

if events is not None:
    # Print the events
    for event in events:
        print(f"Event {event['eventName']}:")
        print(f"  Date: {event['eventDate']}")
        print(f"  Link: {event['eventLink']}")
        print(f"  Image: {event['eventImage']}")
        print()

    # Write the events to a JSON file and store at the root level
    root_dir = os.path.dirname(os.path.abspath(__file__))
    with open(os.path.join(root_dir, '../ministry_of_testing_events.json'), 'w') as f:
        json.dump(events, f)

File: scripts/scrapeEvents.js
const axios = require('axios');
const cheerio = require('cheerio');
const fs = require('fs');
const path = require('path');

// Define the URL of the Ministry of Testing events page
const url = 'https://www.ministryoftesting.com/events/list';

// Function to scrape event data
async function scrapeEvents() {
    try {
        // Send a GET request to the webpage
        console.log(`Sending GET request to ${url}`);
        const { data } = await axios.get(url);
        console.log('Received response from the website.');

        // Load the HTML into cheerio to parse and extract data
        const $ = cheerio.load(data);
        console.log('Loaded HTML into cheerio.');

        // Array to hold the scraped event data
        const events = [];

        // Select and loop through each event card - remember to inspect the page to check the name of the section you're looking for
        const eventCards = $('.summary-card');
        console.log(`Found ${eventCards.length} event cards on the page.`);

        eventCards.each((index, element) => {
            // Extract event name
            const eventName = $(element).find('.title').text().trim();
            console.log(`Event ${index + 1}: Name - ${eventName}`);

            // Extract event date
            const eventDate = $(element).find('.fs-6').text().trim();
            console.log(`Event ${index + 1}: Date - ${eventDate}`);

            // Extract event image
            const eventImage = $(element).find('.card-img-top').attr('src');
            console.log(`Event ${index + 1}: Image - ${eventImage}`);

            // Extract event link and construct full URL
            const eventLink = `https://www.ministryoftesting.com${$(element).find('.stretched-link').attr('href')}`;
            console.log(`Event ${index + 1}: Link - ${eventLink}`);

            // Add the event data to the array
            events.push({
                eventName,
                eventDate,
                eventLink,
                eventImage,
            });
        });

        // Check if events array is populated
        if (events.length > 0) {
            // Write the event data to a JSON file at the root level
            const rootDir = path.resolve(__dirname, '../');
            fs.writeFileSync(`${rootDir}/ministry_of_testing_events.json`, JSON.stringify(events, null, 2), 'utf-8');
            console.log('Event data successfully saved to "ministry_of_testing_events.json"');
        } else {
            console.log('No events found to save.');
        }
    } catch (error) {
        console.error(`Error occurred while scraping: ${error.message}`);
    }
}

// Run the scrape function
scrapeEvents();
